test/path.jl

```diff
@@ -85,6 +85,41 @@
     end
     @test relpath(S(joinpath("foo","bar")), S("foo")) == "bar"
 
+    @testset "splitpath" begin
+        @test splitpath(joinpath("a","b","c")) == ["a", "b", "c"]
+        @test splitpath("") == []
```

Comment:
This seems wrong to me. In particular, note that `joinpath()` is an error, not `""`. This seems like it should be `[""]` instead of `[]`; `joinpath("") == ""` so you get the input back as you would want.

Reply:
hmm. Yeah I *think* that makes sense (from the identity viewpoint).

As another perspective, if I ask _"what are the directories and files in this here path: `""`?"_, then _"There aren't any! That is, it's these ones: `[]`"_ seems to me like the right response. Certainly it seems more plausible an answer than _"ah yes, it contains this file: `[""]`!"_.

Put another way, should `length(splitpath(""))` really be the same as `length(splitpath("a.txt"))`?

So I'm not sure what's best there. I guess it seems noteworthy that you can't get `splitpath` to return `[""]`, I just don't know if that's a good or a bad thing. I do agree that having a potential error lurking inside `joinpath(splitpath(p)...)` seems scary...

Reply:
> There aren't any!

This isn't true though: `""` is the path of the current directory. It has a single component which happens to be the empty string, but that's not really relevant. You can replace the empty string in these cases  with `.` and everything should behave the same.

Reply:
Oh! Really? Yeah well, in that case i'm happy.

In that case, I think it does make sense that `length(splitpath(""))` is `1`. So `length(splitpath(p))` can never be `0`, and I think that makes sense. Okay yeah, I'll change this then. Thanks for explaining that!

Reply:
Done.